tools/system_libs.py

```diff
@@ -638,10 +638,13 @@ def add_dir(p):
         from urllib2 import urlopen
       f = urlopen(url)
       data = f.read()
-      open(fullname + '.zip', 'wb').write(data)
+      open(fullname + ('.zip' if not is_tarbz2 else '.tar.bz2'), 'wb').write(data)
       State.retrieved = True
 
     def check_tag():
+      if is_tarbz2:
```

Comment:
this is necessary, without it you may use an old version even though there is a newer one, which can be very confusing.

Reply:
We should also test that that works. There is a test in `test_sanity.py` for the current zipfiles, so we'd need to add testing for this to there.

Or, could we just use zipfiles for this too?